docs/api/spec/generate-openapi-from-routes.pl

```diff
@@ -0,0 +1,402 @@
+                }
+            },
+            "required" => true,
+        },
+        "responses" => {
+            "201" => {
+                "\$ref" => "#/components/responses/Created"
+            },
+            "400" => {
+                "\$ref" => "#/components/responses/BadRequest"
+            },
+            "422" => {
+                "\$ref" => "#/components/responses/UnprocessableEntity"
+            }
+        },
+    };
+}
+
+sub configCollectionPost {
+    my ($method, $child, $forms) = @_;
+    {
+        "parameters" => [
+            standardParameters()
+        ],
+        "requestBody" => {
+            "content" => {
+                "application/json" => {
+                    configCollectionPostJsonSchema($method, $child, $forms)
+                }
+            },
+            "required" => true,
+        },
+        "responses" => {
+            "201" => {
+                "\$ref" => "#/components/responses/Created"
+            },
+            "400" => {
+                "\$ref" => "#/components/responses/BadRequest"
+            },
+            "409" => {
+                "\$ref" => "#/components/responses/Duplicate"
+            },
+            "422" => {
+                "\$ref" => "#/components/responses/UnprocessableEntity"
+            }
+        },
+    };
+}
+
+sub configCollectionGet {
+    my ( $method, $child, $forms ) = @_;
+    {
+        "parameters" => [
+            standardParameters(),
+            { "\$ref" => '#/components/parameters/cursor' }
+        ],
+        "responses" => {
+            "400" => {
+                "\$ref" => "#/components/responses/BadRequest"
+            },
+            "422" => {
+                "\$ref" => "#/components/responses/UnprocessableEntity"
+            },
+            '200' => {
+                "description" => "A list of connection profiles",
+                "content"     => {
+                    "application/json" => {
+                        "schema" => {
+                            "allOf" => [
+                                {
+                                    "\$ref" => "#/components/schemas/Iterable",
+                                },
+                                {
+                                    'type' => 'object',
+                                    'properties' => {
+                                        items => pf::UnifiedApi::GenerateSpec::formsToSchema($forms),
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+        },
+    };
+}
+
+sub configResourceDelete {
+    {
+        summary     => 'Delete a config item',
+        "parameters" => [
+            standardParameters()
+        ],
+        'responses' => {
+            '204' => {
+                description => 'Deleted a config item'
+            }
+        }
+    }
+}
+
+sub baseInfo {
```

Comment:
IMO, it would be better if we still use the openapi-base.yaml to define those parameters as its more user friendly to write them directly in the YAML instead of in the code here